### PR TITLE
fix public audio style

### PR DESCRIPTION
**Pull request by szaimen**
Follow-up to https://github.com/nextcloud/server/pull/34051

| Before | After |
|---|---|
| image | image |

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: apps/files_sharing/css/public.scss [hunk 1]
```diff
@@ -45,6 +45,14 @@ $download-button-section-height: 200px;
 	margin-right: auto;
 }
 
+#imgframe:has(audio) {
```
Comment (szaimen) [inline apps/files_sharing/css/public.scss:48]: this is such a useful selector 😍

**Review by Pytal — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by szaimen** [comment]
CI failure unrelated

File: apps/files_sharing/css/public.scss (hunk 1 above)
Comment (pulsejet) [reply apps/files_sharing/css/public.scss:48]: Just a note that firefox doesn't support this yet
https://caniuse.com/?search=has

Comment (szaimen) [reply apps/files_sharing/css/public.scss:48]: This is unfortunate! Do you have an idea for a better selector that works everywhere or maybe does @skjnldsv has one?

Comment (max-nextcloud) [reply apps/files_sharing/css/public.scss:48]: I think you could also apply the margin on the `.plyr` element. That only shows when there's an audio file.

Comment (szaimen) [reply apps/files_sharing/css/public.scss:48]: > I think you could also apply the margin on the `.plyr` element. That only shows when there's an audio file.

No, it also shows if there is a video where we do not want a margin... So what could be a solution to this?

Comment (max-nextcloud) [reply apps/files_sharing/css/public.scss:48]: We can add a `data-handler` property to the `#viewer` element just like [we do in the modal](https://github.com/nextcloud/viewer/blob/master/src/views/Viewer.vue#L49).

That will allow styling depending on the selected handler (audio, video, image, text, ...)

Comment (max-nextcloud) [reply apps/files_sharing/css/public.scss:48]: https://github.com/nextcloud/viewer/pull/1371

**Pull request merged**